backend/open_webui/grounding/context_analysis.py

```
@@ -0,0 +1,598 @@
+            r"\b(what|why|how)\s+(about|regarding|concerning)\s+(this|that|it|them)\b",
+            r"\bwhat\s+(triggered|caused|led|made)\b",
+            r"\bhow\s+(did|does)\s+(he|she|it|they)\b",
+            # French patterns
+            r"\b(qui|que|quoi|quand|où|pourquoi|comment)\s+(est|était|sont|étaient|a|ont)\s+(il|elle|ils|elles|ça|cela)\b",
+            r"\b(quoi|pourquoi|comment)\s+(sur|concernant)\s+(ceci|cela|ça)\b",
+            r"\bqu\'est-ce\s+qui\s+(a\s+(déclenché|causé|mené))\b",
+        ]
+
+        return any(re.search(pattern, query_lower) for pattern in question_patterns)
+
+        return "\n".join(context_parts[-5:])  # Last 5 context entries
```

Ruff (0.12.2)

Undefined name `context_parts`

(F821)